src/ogdc_runner/recipe/viz_workflow.py

```diff
@@ -0,0 +1,319 @@
+from __future__ import annotations
+
+from hera.workflows import (
+    DAG,
+    Artifact,
+    HTTPArtifact,
+    Parameter,
+    Workflow,
+    script,
+)
+from hera.workflows.models import (
+    PersistentVolumeClaimVolumeSource,
+    Volume,
+    VolumeMount,
+)
+from loguru import logger
+from pydantic import AnyUrl
+
+from ogdc_runner.argo import (
+    ARGO_WORKFLOW_SERVICE,
+    submit_workflow,
+)
+
+# Import common utilities
+from ogdc_runner.common import (
+    apply_custom_container_config,
+    parse_commands_from_recipe_file,
+)
+from ogdc_runner.constants import SIMPLE_RECIPE_FILENAME
+from ogdc_runner.models.recipe_config import RecipeConfig
+from ogdc_runner.recipe import get_recipe_config
+
+
+@script(
+    name="download-viz-config",
+    outputs=[
+        Artifact(name="viz-config-json", path="/mnt/workflow/config/viz-config.json"),
+    ],
+    image="python:3.10-slim",
+    command=["python"],
+    volume_mounts=[
+        VolumeMount(name="qgnet-ogdc-workflow-pvc", mount_path="/mnt/workflow")
+    ],
+)
+def download_viz_config(config_url) -> None:  # type: ignore[no-untyped-def]
+    """Downloads visualization configuration from a URL."""
+    import subprocess
+    import sys
+
+    # Install requests
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "requests"])
+
+    from pathlib import Path
+
+    import requests  # type: ignore[import-untyped]
+
+    output_path = Path("{{outputs.artifacts.viz-config-json.path}}")
+    output_path.parent.mkdir(parents=True, exist_ok=True)
+
+    response = requests.get(config_url)
+    response.raise_for_status()
+
+    with Path.open(output_path, "w") as f:
+        f.write(response.text)
+
+
+@script(
+    name="batching",
+    inputs=[
+        Parameter(name="input_url"),
+        HTTPArtifact(
+            name="batch-input",
+            path="/mnt/workflow/input/ice_basins.gpkg",
```

Comment:
This is specific to a particular dataset and should be made more generic for use with other datasets.